### PR TITLE
[NO-JIRA] Fix flow for infinite scroll

### PR DESCRIPTION
**Pull request by tiagoengel**
I've introduced a test file for flow here, the idea is to test if required and default props are working as expected.

Lines with the `$ExpectError` annotation will suppress the error in the next line, if any, or show a warning saying there are no errors in the next line, which means the flow typing is wrong as we are expecting an error.

Because it only throw a warning in this case I have also changed the build to fail if any warnings are shown, so if we break the flow typing in any way the build will fail.

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack/blob/master/CONTRIBUTING.md)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: packages/bpk-component-infinite-scroll/src/withInfiniteScroll.js [hunk 1]
```diff
@@ -18,7 +18,12 @@
 
 /* @flow */
```
Comment (ojcurtis) [inline packages/bpk-component-infinite-scroll/src/withInfiniteScroll.js]: Can you make this `/* @flow strict */`?

File: packages/bpk-component-infinite-scroll/stories.js [hunk 2]
```diff
@@ -16,6 +16,8 @@
  * limitations under the License.
  */
 
+/* @flow */
```
Comment (ojcurtis) [inline packages/bpk-component-infinite-scroll/stories.js]: Can you make this /* @flow strict */?

File: packages/bpk-component-infinite-scroll/src/withInfiniteScroll-test.flow.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    <InfiniteList aria-label="infinite list" />
+
+    {/* $ExpectError (incompatible type) */}
+    <InfiniteList dataSource={new ArrayDataSource([])} aria-label={null} />
+
+    {/* Test default props */}
+    <InfiniteList
+      dataSource={new ArrayDataSource([])}
+      aria-label="infinite list"
+    />
```
Comment (shaundon) [inline packages/bpk-component-infinite-scroll/src/withInfiniteScroll-test.flow.js]: I'm a bit confused as to how this tests default props. I get that when this is rendered the default props will be added, but what are we doing to verify that in this test file?

Comment (tiagoengel) [reply packages/bpk-component-infinite-scroll/src/withInfiniteScroll-test.flow.js]: The idea is, if default props are not working we will get a flow error here, so as long as we don't get any error here it means flow is considering default props correctly, or so I hope 😅

Comment (shaundon) [reply packages/bpk-component-infinite-scroll/src/withInfiniteScroll-test.flow.js]: Ahh gotcha

**Pull request merged**